gfx-memory/src/heaps/mod.rs

```diff
@@ -165,7 +165,14 @@ impl<B: hal::Backend> Heaps<B> {
             return Err(hal::device::OutOfMemory::Device.into());
         }
 
-        let (flavor, allocated) = memory_type.alloc(device, kind, size, align)?;
+        let (flavor, allocated) = match memory_type.alloc(device, kind, size, align) {
+            Ok(mapping) => mapping,
+            Err(e) if kind != Kind::Dedicated => {
```

Comment:
Is it intentional that we also fallback to `Kind::Dedicated` for `Kind::General` here too?

Reply:
Yeah, come to think of it, Dedicated has its own way of falling back. I pushed a version that only checks for Linear here